src/plugins/intel_npu/src/plugin/npuw/just_sync_infer_request.cpp

```diff
@@ -579,8 +646,9 @@ void ov::npuw::JustInferRequest::bind_global_parameters(std::size_t idx) {
         // Otherwise: Just a return a subrequest which is in place.
         // If it is a function call and we have function pipelining ON,
         // it is still the right subrequest we can use.
-        LOG_DEBUG("Accessing the primary subrequest");
-        return m_subrequests[real_idx];
+        auto request_idx = get_request_id_for_submodel(idx);
```

Comment:
@AsyaPronina this changes required for past-key-values tensors to be properly connected to inferrequests